src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1719,10 +1698,9 @@ private IEnumerable<TypeInferenceInfo> InferTypeInNameColon(NameColonSyntax name
                                     ? delegateType.TypeArguments[0]
                                     : this.Compilation.ObjectType;
 
-                                if (IsUnusableType(typeArg) && argumentExpression is LambdaExpressionSyntax)
+                                if (IsUnusableType(typeArg) && argumentExpression is LambdaExpressionSyntax lambdaExpression)
```

Comment:
📝 This one should have been caught by our existing analyzer. Can you file a bug if it was not?

Reply:
will check!